### PR TITLE
[Version] v1.6.23

### PR DESCRIPTION
**Pull request by Herklos**
### Updated
- Migrate order tests
- Refactor book data class

### Removed
- Pandas requirement

### CODE REVIEW

File: tests/data/test_order.py [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+    max_price = 10.454677
+    min_price = 2.4273
+    order_inst.creation_time = time.time()
+    recent_trades = [{"price": random_price(min_value=min_price, max_value=max_price), "timestamp": time.time()}
+                     for _ in range(0, SIMULATOR_LAST_PRICES_TO_CHECK)]
+
+    # append validating trade
+    recent_trades.append({"price": max_price, "timestamp": time.time()})
+    assert order_inst.check_last_prices(recent_trades,
+                                        random_price(min_value=min_price, max_value=max_price - 1),
```
Comment (GuillaumeDSM) [inline tests/data/test_order.py:110]: won't this make random failures sometimes ?

File: tests/data/test_order.py [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+    order_inst.creation_time = time.time()
+    recent_trades = [{"price": random_price(min_value=min_price, max_value=max_price), "timestamp": time.time()}
+                     for _ in range(0, SIMULATOR_LAST_PRICES_TO_CHECK)]
+
+    assert not order_inst.check_last_prices(recent_trades,
+                                            max_price,
+                                            inferior=False)
+
+
+async def test_new(trader):
```
Comment (GuillaumeDSM) [inline tests/data/test_order.py:135]: it looks like it's more of a ```test_update``` than ```test_new```

File: tests/data/test_order.py [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+    assert order_inst.status == OrderStatus.OPEN
+
+    order_inst.update(order_type=TraderOrderType.STOP_LOSS_LIMIT,
+                      symbol="ETH/BTC",
+                      quantity=0.1,
+                      quantity_filled=5.2,
+                      price=0.12,
+                      stop_price=0.9)
+    assert order_inst.origin_stop_price == 0.9
+    assert order_inst.origin_price == 0.12
```
Comment (GuillaumeDSM) [inline tests/data/test_order.py:161]: should we also tests other computed fields ? like filled_quantity, exchange_order_type, filled_price and executed_time

File: tests/traders/__init__.py [hunk 4]
```diff
@@ -13,3 +13,30 @@
...
+    return config, exchange_manager, trader_inst
+
+
+@pytest.fixture
+async def trader_simulator(exchange_manager):
+    config = load_test_config()
+    config[CONFIG_SIMULATOR][CONFIG_ENABLED_OPTION] = True
+    trader_inst = TraderSimulator(load_test_config(), exchange_manager)
+    await trader_inst.initialize()
+    return config, exchange_manager, trader_inst
```
Comment (GuillaumeDSM) [inline tests/traders/__init__.py:42]: 👍

File: tests/data/test_order.py (hunk 1 above)
Comment (Herklos) [reply tests/data/test_order.py:110]: It was already a random generation. I'll improve that

File: tests/data/test_order.py (hunk 3 above)
Comment (Herklos) [reply tests/data/test_order.py:161]: You want to test default values or set other values to update ?

Comment (GuillaumeDSM) [reply tests/data/test_order.py:161]: The ones I listed are computed values during ```update```, I don't think default values are worth testing but computed ones we should test

**Pull request merged**